### PR TITLE
auth: enforce max NIP-05 username length boundary

### LINKED ISSUE #80: Follow-up: add NIP-05 username length limit
Follow-up from PR #72.

`normalize_nip05_username` currently has no explicit length bound. The database column may constrain this eventually, but validation should reject oversized usernames before hitting storage or downstream integrations.

Please add an explicit maximum length, such as 64 characters, and tests for the boundary.

### PR DESCRIPTION
**Pull request by NotThatKindOfDrLiz**
Closes #80.

This is a clean recut of the actual follow-up work from #99 on top of current `main`.

What it does:
- adds a max length boundary for normalized NIP-05 usernames
- adds unit coverage for the exact boundary and the first-over-boundary case

Why this exists as a fresh PR:
- the original #99 branch was stacked on pre-#72 NIP-05 work that is already merged
- this branch contains only the one unique follow-up commit on top of current `main`

### CODE REVIEW

File: api/src/api/http/auth.rs [hunk 1]
```diff
@@ -92,6 +93,13 @@ fn normalize_nip05_username(raw_username: &str) -> Result<String, AuthError> {
         ));
     }
 
+    if username.len() > MAX_NIP05_USERNAME_LENGTH {
+        return Err(AuthError::Internal(format!(
```
Comment (dcadenas) [inline api/src/api/http/auth.rs:97]: The new length check returns `AuthError::Internal`, but that variant's `IntoResponse` impl at auth.rs:376-383 logs at ERROR and sends HTTP 503 with a generic "Service temporarily unavailable" message. A user who posts a 65-character username will get a 503 and never see the "Username must be at most 64 characters" text in the error itself. The other guards in `normalize_nip05_username` already do this, so the new branch is consistent with the surrounding code. But since you're adding a branch here anyway, `AuthError::BadRequest` would fit the semantics better: this is a client input violation, not a server-internal failure, and `BadRequest` would actually deliver the message string to the caller.

**Review by dcadenas — APPROVED** [review]
🔒 Good to have the upper bound pinned down on NIP-05 usernames.
